### PR TITLE
Improve scoping of nested paragraph right-padding rule

### PR DESCRIPTION
**Pull request by jasmussen**
This addresses feedback in https://github.com/WordPress/gutenberg/pull/11757#issuecomment-440269586.

The initial issue was that in the default appender, and in empty paragraphs, the plus to add blocks overlaps the placeholder text. The right-padding fixed that. But it bled into other blocks with placeholders, such as the Button block.

This PR scopes the rich text rule to only apply to empty paragraphs. The appender was already scoped.

Before:

before

After:

after

The button isn't as wide on the right side, in the after.

### CODE REVIEW

File: packages/editor/src/components/rich-text/style.scss [hunk 1]
```diff
@@ -103,7 +103,7 @@
 
 		// On mobile and in nested contexts, the plus to add blocks shows up on the right.
 		// This padding makes sure it doesn't overlap text.
-		& + .editor-rich-text__tinymce {
+		& + .editor-rich-text__tinymce.wp-block-paragraph {
```
Comment (youknowriad) [inline packages/editor/src/components/rich-text/style.scss]: I don't like that we target specific blocks in a generic component, what's the reason here? We could do the opposite if needed. Target the richText in the block.

Comment (jasmussen) [reply packages/editor/src/components/rich-text/style.scss]: >  Target the richText in the block.

Do you mean move these styles into the paragraph block? 

Given we have a unique case for the _empty_ paragraph block, I think it's okay to make rules specific to that. But you could have a point in where the styles should live.

Comment (youknowriad) [reply packages/editor/src/components/rich-text/style.scss]: yes, move to paragraph what's specific to paragraph and leave here what's generic to RichText.

Comment (jasmussen) [reply packages/editor/src/components/rich-text/style.scss]: Good thoughts. Fixed it in https://github.com/WordPress/gutenberg/pull/12106/commits/0879d8a81d8370c4ee7b750717d6d4fab33ec9cb:

screenshot 2018-11-20 at 15 35 46

File: packages/block-library/src/paragraph/style.scss [hunk 2]
```diff
@@ -37,3 +37,10 @@ p.has-background {
 p.has-text-color a {
 	color: inherit;
 }
+
+// Specific to the empty paragraph placeholder:
+// when shown on mobile and in nested contexts, the plus to add blocks shows up on the right.
+// This padding makes sure it doesn't overlap text.
+.editor-rich-text__tinymce[data-is-placeholder-visible="true"] + .editor-rich-text__tinymce.wp-block-paragraph {
```
Comment (youknowriad) [inline packages/block-library/src/paragraph/style.scss]: It should be in `editor.scss` right?

Comment (jasmussen) [reply packages/block-library/src/paragraph/style.scss]: Argh, yes it should. Many apologies, I know you're busy. https://github.com/WordPress/gutenberg/pull/12106/commits/81a81a47a9d3b3ecb38baadcd85a3de020df95ca should fix that up.

**Comment by catehstn** [comment]
@youknowriad looks like your comments have been addressed -- can we get this merged?

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Pull request merged**